src/SFML/Window/OSX/CursorImpl.mm

```diff
@@ -71,7 +71,11 @@
 ////////////////////////////////////////////////////////////
 bool CursorImpl::loadFromPixels(const Uint8* pixels, Vector2u size, Vector2u hotspot)
 {
-    [m_cursor release];
+    if (m_cursor)
+    {
+        [m_cursor release];
+        m_cursor = nil;
+    }
```

Comment:
If one had a valid cursor before and the loading would fail for some reason, the user would end up with a different state.

I guess it can be argued either way. For an UX perspective, I think it would be nicer to retain the valid cursor if the loading failed, but I'm open either way.

Reply:
Yeah in many ways that would be user error - if the load function fails then you shouldn't use the cursor (same as with other resources like audio/textures) but I do agree that it makes sense for the cursor object to maintain it's state if loading fails.

Think the only way to do that is to make a temp local copy of m_cursor which is released at the end of the method - will update this later